system/lib/wasmfs/syscalls.cpp

```diff
@@ -837,8 +845,7 @@ long __syscall_getdents64(long fd, long dirp, long count) {
     result->d_reclen = sizeof(dirent);
     result->d_type =
       curr.file->is<Directory>() ? DT_DIR : DT_REG; // TODO: add symlinks.
-    // TODO: Enforce that the name can fit in the d_name field.
-    assert(curr.name.size() + 1 <= sizeof(result->d_name));
+    assert(curr.name.size() < sizeof(result->d_name));
```

Comment:
I think the ` + 1` was there to account for the null terminator, which won't be counted in `curr.name.size()`. That seems important to keep.

Reply:
Is this just for semantics? (i.e. so that someone knows that the they have to consider the null terminator?) I think this is the same either way.

Reply:
Since there are no fractional values, this statement is still true (but a little stronger). I guess we could keep the `+1` so that readers are aware of the null terminator?